scripts/tests-coverage-diff.ts

```diff
@@ -0,0 +1,13 @@
+import { diff as coverageDiff } from 'coverage-diff';
+import { readFileSync, writeFileSync } from 'fs';
+
+const getDiff = () => {
+  const base = JSON.parse(readFileSync('coverage-master/coverage-summary.json').toString());
+  const head = JSON.parse(readFileSync('coverage/report/coverage-summary.json').toString());
+  return coverageDiff(base, head);
+};
+
+(() => {
+  const { results } = getDiff();
+  writeFileSync('coverage/report/coverage-diff.txt', results, { flag: 'w' });
+})();
```

Comment:
The purpose of this script is to produce a coverage report that can be used in a PR. The test workflow now runs this script, retrieves the diff text and adds a comment to the PR via a new action.

Reply:
I don't know if you've seen this, but I just stumbled upon this other approach:
 - https://stackoverflow.com/a/77008544

It uses another GH action. Not sure if it's worth a try, probably not.

I'm sharing just in case.

Reply:
The action mentioned in the thread does not support coverage comparison, so we couldn't produce a report based off 2 separate runs (master vs PR branch). Another motivation for this coverage diff script.

Reply:
Certainly not worth it, then. 😅